src/safeds/data/tabular/transformation/_k_nearest_neighbors_imputer.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from numpy import nan
+
+from safeds._utils import _structural_hash
+from safeds._validation import _check_bounds, _check_columns_exist, _ClosedBound
+from safeds.data.tabular.containers import Table
+from safeds.exceptions import TransformerNotFittedError
+
+from ._table_transformer import TableTransformer
+
+if TYPE_CHECKING:
+    from sklearn.impute import KNNImputer as sk_KNNImputer
+
+
+class KNearestNeighborsImputer(TableTransformer):
+    """
+    The KNearestNeighborsImputer replaces missing values in given Columns with the mean value of the K-nearest neighbors.
+
+    Parameters
+    ----------
+    neighbor_count:
+        The number of neighbors to consider when imputing missing values.
+    column_names:
+        The list of columns used to impute missing values. If 'None', all columns are used.
+    value_to_replace:
+        The placeholder for the missing values. All occurrences of`missing_values` will be imputed.
+    """
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Dunder methods
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def __init__(
+        self,
+        neighbor_count: int,
+        *,
+        column_names: str | list[str] | None = None,
+        value_to_replace: float | str | None = nan,
+    ) -> None:
+        super().__init__(column_names)
+
+        _check_bounds(name="neighbor_count", actual=neighbor_count, lower_bound=_ClosedBound(1))
+
+        # parameter
+        self._neighbor_count: int = neighbor_count
+        self._value_to_replace: float | str | None = value_to_replace if value_to_replace is not None else nan
+
+        # attributes
+        self._wrapped_transformer: sk_KNNImputer | None = None
+
+    def __hash__(self) -> int:
+        return _structural_hash(
+            super().__hash__(),
+            self._neighbor_count,
+            self._value_to_replace,
+            # Leave out the internal state for faster hashing
+        )
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Properties
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @property
+    def is_fitted(self) -> bool:
+        """Whether the transformer is fitted."""
+        return self._wrapped_transformer is not None
+
+    @property
+    def neighbor_count(self) -> int:
+        """The number of neighbors to consider when imputing missing values."""
+        return self._neighbor_count
+
```

Comment:
An accessor for `value_to_replace` is missing. Since the user should not get `nan`, you are better off moving the replacement of `None` with `nan` out of `__init__` and into `fit`.